nifi-nar-bundles/nifi-hive-bundle/nifi-hive3-processors/src/main/java/org/apache/nifi/processors/hive/UpdateHive3Table.java

```diff
@@ -0,0 +1,494 @@
+    }
+
+    @Override
+    public void onTrigger(ProcessContext context, ProcessSession session) throws ProcessException {
+
+        FlowFile flowFile = session.get();
+        if (flowFile == null) {
+            return;
+        }
+
+        final RecordReaderFactory recordReaderFactory = context.getProperty(RECORD_READER).asControllerService(RecordReaderFactory.class);
+        final String dbName = context.getProperty(DB_NAME).evaluateAttributeExpressions(flowFile).getValue();
+        final String tableName = context.getProperty(TABLE_NAME).evaluateAttributeExpressions(flowFile).getValue();
+        final String staticPartitionValuesString = context.getProperty(STATIC_PARTITION_VALUES).evaluateAttributeExpressions(flowFile).getValue();
+        List<String> staticPartitionValues = null;
+        if (!StringUtils.isEmpty(staticPartitionValuesString)) {
+            staticPartitionValues = Arrays.stream(staticPartitionValuesString.split(",")).filter(Objects::nonNull).map(String::trim).collect(Collectors.toList());
+        }
+
+        final ComponentLog log = getLogger();
+
+        try {
+            final RecordReader reader;
+
+            try (final InputStream in = session.read(flowFile)) {
+                // if we fail to create the RecordReader then we want to route to failure, so we need to
+                // handle this separately from the other IOExceptions which normally route to retry
+                try {
+                    reader = recordReaderFactory.createRecordReader(flowFile, in, getLogger());
+                } catch (Exception e) {
+                    throw new RecordReaderFactoryException("Unable to create RecordReader", e);
+                }
+            } catch (RecordReaderFactoryException rrfe) {
+                log.error(
+                        "Failed to create {} for {} - routing to failure",
+                        new Object[]{RecordReader.class.getSimpleName(), flowFile},
+                        rrfe
+                );
+                session.transfer(flowFile, REL_FAILURE);
+                return;
+            }
+
+            RecordSchema recordSchema = reader.getSchema();
+
+            final boolean createIfNotExists = context.getProperty(CREATE_TABLE).asBoolean();
+            final String storageFormat = context.getProperty(TABLE_STORAGE_FORMAT).getValue();
+            final Hive3DBCPService dbcpService = context.getProperty(HIVE_DBCP_SERVICE).asControllerService(Hive3DBCPService.class);
+            try (final Connection connection = dbcpService.getConnection()) {
+
+                checkAndUpdateTableSchema(session, flowFile, connection, recordSchema, dbName, tableName, staticPartitionValues, createIfNotExists, storageFormat);
+                Map<String, String> updateAttributes = new HashMap<>();
+                updateAttributes.put(ATTR_OUTPUT_TABLE, dbName + "." + tableName);
+                flowFile = session.putAllAttributes(flowFile, updateAttributes);
```

Comment:
That is copied from some other processor (PutHive3Streaming I think?). It's in two places in case an error occurs during processing, the output flowfile will have that attribute whether it is routed to success or failure. Since there is only one attribute in this case, I will reduce it to a single putAttribute() as you suggest